packages/craco/lib/features/webpack/eslint.js

```diff
@@ -48,80 +48,132 @@ function extendsEslintConfig(loader, eslintConfig, context) {
+        webpackEslintConfig.options = loaderOptions(webpackEslintConfig.options || {}, context);
 
-        if (!loader.options) {
+        if (!webpackEslintConfig.options) {
             throw new Error("craco: 'eslint.loaderOptions' function didn't return a loader config object.");
         }
     } else {
         // TODO: ensure is otherwise a plain object, if not, log an error.
-        loader.options = deepMergeWithArray(loader.options || {}, loaderOptions);
+        webpackEslintConfig.options = deepMergeWithArray(webpackEslintConfig.options || {}, loaderOptions);
     }
 
     log("Applied ESLint loader options.");
 }
 
 function overrideEsLint(cracoConfig, webpackConfig, context) {
     if (cracoConfig.eslint) {
-        const { isFound, match } = getLoader(webpackConfig, loaderByName("eslint-loader"));
+        const { isFound, webpackEslintConfig, disableEsLint } = loadWebpackEslintConfig(cracoConfig, webpackConfig);
 
         if (!isFound) {
-            logError("Cannot find ESLint loader (eslint-loader).");
-
             return webpackConfig;
         }
 
         const { enable, mode, loaderOptions } = cracoConfig.eslint;
 
         if (enable === false) {
-            disableEslint(webpackConfig);
+            const { hasRemovedAny } = disableEsLint();
+
+            if (hasRemovedAny) {
+                log("Disabled ESLint.");
+            } else {
+                logError("Couldn't disabled ESLint.");
+            }
 
             return webpackConfig;
         }
 
-        enableEslintIgnoreFile(match.loader);
+        enableEslintIgnoreFile(webpackEslintConfig);
 
         if (mode === ESLINT_MODES.file) {
-            useEslintConfigFile(match.loader);
+            useEslintConfigFile(webpackEslintConfig);
         } else {
-            extendsEslintConfig(match.loader, cracoConfig.eslint, context);
+            extendsEslintConfig(webpackEslintConfig, cracoConfig.eslint, context);
         }
 
         if (loaderOptions) {
-            applyLoaderOptions(match.loader, loaderOptions);
+            applyLoaderOptions(webpackEslintConfig, loaderOptions);
         }
     }
 
     return webpackConfig;
 }
 
+function loadWebpackEslintConfig(cracoConfig, webpackConfig) {
+    let webpackEslintConfig = {
+        options: null
+    };
+
+    let disableEsLint = null;
+
+    // The eslint-loader has been deleted from create-react-app in version 4.0. Now eslint is being delivered through eslint-webpack-plugin, not eslint-loader.
+    // https://github.com/facebook/create-react-app/commit/d07b7d025f5933710fcb01718617dbdf4bc54c33
+    if (semver.gte(getReactScriptVersion(cracoConfig), "4.0.0")) {
+        const matcher = pluginByName("ESLintWebpackPlugin");
+        const { isFound, match } = getPlugin(webpackConfig, matcher);
+
+        if (!isFound) {
+            logError("Cannot find ESLint plugin (ESLintWebpackPlugin).");
+
+            return { isFound: false };
+        }
+
+        webpackEslintConfig = {
```

Comment:
I see what you did here... I guess it works but I wonder if instead this function should return an "updateEslintConfig(options)" instead?

Still I don't mind keeping this, it's less obvious though the first time you run the code and you end up having "if" that apply only to the old loader structure in every function (extendsEslintConfig, useEslintConfigFile, enableEslintIgnoreFile)

Reply:
I initially did something similar, but it did not feel good. 

loadWebpackEslintConfig would return 

```
    return {
        isFound: true,
        updateEslintConfig: options => update(options),
        disableEsLint
    };
```


Then instead of having methods like this : 
```

function useEslintConfigFile(webpackEslintConfig) {
    if (webpackEslintConfig.options) {
        webpackEslintConfig.options.useEslintrc = true;
        delete webpackEslintConfig.options.baseConfig;
    } else {
        webpackEslintConfig.options = {
            useEslintrc: true
        };
    }

    log("Overrided ESLint config to use a config file.");
}
```


i would end up with

```

function useEslintConfigFile(updateEslintConfig) {
    updateEslintConfig({ useEslintrc: true });
        
   // then i would need an extra param in updateEslintConfig for this:  delete webpackEslintConfig.options.baseConfig;
   // updateEslintConfig(options, deleteBaseConfig)?
    log("Overrided ESLint config to use a config file.");
}
```